src/main/java/me/karakelley/http/filesystem/PublicDirectory.java

```diff
@@ -97,4 +93,24 @@ public void updateFileContents(String path, byte[] contents) {
       throw new RuntimeException(e);
     }
   }
+
+  private boolean isValidPath(String requestedPath) {
+    File canonicalPath = normalizeFullPath(requestedPath);
+
+    return canonicalPath.toString().contains(normalizedDocumentRoot().toString());
```

Comment:
It feels strange to be checking if a path is valid by doing a string comparison. Can you use [this](https://docs.oracle.com/javase/7/docs/api/java/nio/file/Path.html#startsWith(java.nio.file.Path))?

Reply:
So if I pass in the directory `./thispath/somewhere` the canonical path is something like `/Users/karakelley/code/thispath/somewhere` and `normalizedDocumentRoot` would be `/thispath/somewhere` which means the canonical root would never start with `normalizedDocumentRoot` (because its missing part of the full path). I really just want to see if the full document root is included within the requested path. I can see if there might be another alternative to string comparison but so far not seeing anything in the `Path` interface.